volume/local/local_unix.go

```diff
@@ -112,18 +113,33 @@ func (v *localVolume) mount() error {
 	if v.opts.MountDevice == "" {
 		return fmt.Errorf("missing device in volume options")
 	}
+
 	mountOpts := v.opts.MountOpts
+	mountDevice := v.opts.MountDevice
+
 	switch v.opts.MountType {
-	case "nfs", "cifs":
+	case "nfs":
 		if addrValue := getAddress(v.opts.MountOpts); addrValue != "" && net.ParseIP(addrValue).To4() == nil {
 			ipAddr, err := net.ResolveIPAddr("ip", addrValue)
 			if err != nil {
 				return errors.Wrapf(err, "error resolving passed in network volume address")
 			}
 			mountOpts = strings.Replace(mountOpts, "addr="+addrValue, "addr="+ipAddr.String(), 1)
 		}
+	case "cifs":
+		deviceUrl, err := url.Parse(v.opts.MountDevice)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing mount device url")
+		}
+		if deviceUrl.Host != "" && net.ParseIP(deviceUrl.Host).To4() == nil {
```

Comment:
```suggestion
		if deviceUrl.Host != "" && net.ParseIP(deviceUrl.Host) == nil {
```

I know this was just taken from the previous code, but I'm wondering if there is a reason to restrict the lookup to IPv4 addresses only?

Reply:
Original PR for nfs was https://github.com/moby/moby/pull/27329

I don't see an immediate mention during review about IPv4, but there was a `IsIP()` helper (that was later removed?) https://github.com/moby/moby/pull/27329#discussion_r83203368

And I saw some mention of a complex RegEx that was later removed, so I wonder if the `.To4()` was just use as validation for "is it an IP address?")

We should look if we can use IPv6 addresses without issues (or if they need the bracketed (`[::1]`) notation or escaping)